python/cudf/dataframe/multiindex.py

```diff
@@ -378,6 +378,31 @@ def to_frame(self, index=True, name=None):
             df.columns = name
         return df
 
+    def get_level_values(self, level):
+        from cudf import DataFrame
+        colnames = list(self.codes.columns)
+        if level not in colnames:
+            if isinstance(level, int):
+                if level < 0:
+                    level = level - len(colnames)
+                if level < 0 or level >= len(colnames):
+                    raise IndexError(f"Invalid level number: '{level}'")
+                level_idx = level
+                level = colnames[level_idx]
+            else:
+                raise KeyError(f"Level not found: '{level}'")
+        else:
+            level_idx = list(self.codes.columns).index(level)
+        col = self.codes[level].copy()
```

Comment:
This defaults to deep copy and is expensive. Columns are immutable anyway, so why make a copy?

Reply:
Great catch